insights/src/main/java/org/jboss/as/telemetry/extension/InsightsExtension.java

```diff
@@ -0,0 +1,116 @@
+    public static final String PROXY_USER = "proxyUser";
+    public static final String PROXY_PASSWORD = "proxyPassword";
+    public static final String PROXY_URL = "proxyUrl";
+    public static final String PROXY_PORT = "proxyPort";
+    protected static final String TYPE = "insightsType";
+    protected static final PathElement TYPE_PATH = PathElement.pathElement(TYPE);
+
+    /**
+     * The name space used for the {@code subsystem} element
+     */
+    public static final String NAMESPACE = "urn:org.jboss.as.insights:1.0";;
+
+    /**
+     * The name of our subsystem within the model.
+     */
+    public static final String SUBSYSTEM_NAME = "insights";
+
+    /**
+     * Version of the subsystem used for registering
+     */
+    private static final int MAJOR_VERSION = 1;
+
+    /**
+     * The parser used for parsing our subsystem
+     */
+    private final SubsystemParser parser = new SubsystemParser();
+
+    protected static final PathElement SUBSYSTEM_PATH = PathElement.pathElement(SUBSYSTEM, SUBSYSTEM_NAME);
+    private static final String RESOURCE_NAME = InsightsExtension.class.getPackage().getName() + ".LocalDescriptions";
+
+    static StandardResourceDescriptionResolver getResourceDescriptionResolver(final String keyPrefix) {
+        String prefix = SUBSYSTEM_NAME + (keyPrefix == null ? "" : "." + keyPrefix);
+        return new StandardResourceDescriptionResolver(prefix, RESOURCE_NAME, InsightsExtension.class.getClassLoader(), true, true);
+    }
+
+    @Override
+    public void initializeParsers(ExtensionParsingContext context) {
+        context.setSubsystemXmlMapping(SUBSYSTEM_NAME, NAMESPACE, parser);
+    }
+
+    @Override
+    public void initialize(ExtensionContext context) {
+        final SubsystemRegistration subsystem = context.registerSubsystem(SUBSYSTEM_NAME, ModelVersion.create(MAJOR_VERSION));
+        final ManagementResourceRegistration registration = subsystem.registerSubsystemModel(InsightsSubsystemDefinition.INSTANCE);
+        registration.registerOperationHandler(GenericSubsystemDescribeHandler.DEFINITION, GenericSubsystemDescribeHandler.INSTANCE);
+        if (context.isRuntimeOnlyRegistrationValid()) {
+            registration.registerOperationHandler(InsightsRequestHandler.DEFINITION, InsightsRequestHandler.INSTANCE);
+        }
+        subsystem.registerXMLElementWriter(parser);
+    }
+
+    /**
+     * The subsystem parser, which uses stax to read and write to and from xml
+     */
+    private static class SubsystemParser implements XMLStreamConstants, XMLElementReader<List<ModelNode>>, XMLElementWriter<SubsystemMarshallingContext> {
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void readElement(XMLExtendedStreamReader reader, List<ModelNode> list) throws XMLStreamException {
+            // Require no attributes
+            ParseUtils.requireNoAttributes(reader);
+            ParseUtils.requireNoContent(reader);
```

Comment:
subsystem parser that doesn't parse any attributes that are defined in the subsystem?
that is wrong.